Scripts/Runtime/Job/AccessControl/AccessType.cs

```diff
@@ -10,4 +12,37 @@ public enum AccessType
         SharedRead,
         Disposal
     }
-}
+
+    /// <summary>
+    /// A collection of extensions for <see cref="AccessType"/>.
+    /// </summary>
+    public static class AccessTypeExtension
+    {
+        /// <summary>
+        /// Identifies whether the <see cref="AccessType"/> accommodates the operations of the supplied
+        /// <see cref="targetType"/>.
+        /// </summary>
+        /// <param name="type">The type to check for support of the other type's operations.</param>
+        /// <param name="targetType">The type to check for support of.</param>
+        /// <returns>True if <see cref="type"/> supports the operations required by <see cref="targetType"/>.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown if the access type is not yet supported by this method.
+        /// </exception>
+        /// <example>
+        /// An <see cref="AccessType"/> of <see cref="AccessType.SharedWrite"/> supports all of the operations needed
+        /// for <see cref="AccessType.SharedRead"/> access.
+        /// </example>
+        public static bool IsCompatibleWith(this AccessType type, AccessType targetType)
+        {
+            return type == targetType
+                || type switch
```

Comment:
I think I got this logic write but a second brain to think through what operations each AccessType can support would be helpful.

Reply:
Yes, I went through it and the logic is correct.